Framework/DataObjects/test/PeakTest.h

```diff
@@ -363,6 +375,124 @@ class PeakTest : public CxxTest::TestSuite {
     TS_ASSERT_EQUALS(p2.getDetectorID(), -1);
   }
 
+  void test_setQSampleFrameVirtualDetectorWithQLab() {
+    constexpr auto radius = 10.;
+    auto sphereInst =
+        ComponentCreationHelper::createTestInstrumentRectangular(5, 100);
+    auto extendedSpaceObj =
+        ComponentCreationHelper::createSphere(10., V3D(0, 0, 0));
+    auto extendedSpace = new ObjComponent("extended-detector-space",
+                                          extendedSpaceObj, sphereInst.get());
+    extendedSpace->setPos(V3D(0.0, 0.0, 0.0));
+    sphereInst->add(extendedSpace);
```

Comment:
Would using a `unique_ptr` for and `release()` here help to see the intended transfer of ownership more easily?